glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt

```diff
@@ -187,9 +198,11 @@ internal class MetricsPingScheduler(
         // schedule the metrics ping for immediate collection. We only need to perform
         // this check at startup (when overduePingAsFirst is true).
         if (isDifferentVersion()) {
+            Log.i(LOG_TAG, "The application just updated. Send metrics ping now.")
+
             @Suppress("EXPERIMENTAL_API_USAGE")
             Dispatchers.API.executeTask {
-                collectPingAndReschedule(now, startupPing = true)
+                collectPingAndReschedule(now, startupPing = true, reason = Pings.metricsReasonCodes.upgrade)
```

Comment:
Let's also update [this doc](https://mozilla.github.io/glean/book/user/pings/metrics.html) mentioning the reasons. We should probably also cross-check if the list of edge cases is still correct and potentially add the reasons there, too.